adam/curriculum_to_html.py

```diff
@@ -0,0 +1,184 @@
+import os
+from typing import List
+
+from attr import attrs
+
+from adam.experiment import InstanceGroup
+from adam.language.dependency import LinearizedDependencyTree
+from adam.perception import PerceptualRepresentation
+from adam.perception.developmental_primitive_perception import (
+    DevelopmentalPrimitivePerceptionFrame,
+)
+from adam.situation.high_level_semantics_situation import HighLevelSemanticsSituation
+
+
+@attrs(frozen=True, slots=True)
+class CurriculumToHtml:
+    """
+    Class to turn an `InstanceGroup` into an html document
+    """
+
+    @staticmethod
+    def generate(
+        instances: List[
+            InstanceGroup[
+                HighLevelSemanticsSituation,
+                LinearizedDependencyTree,
+                DevelopmentalPrimitivePerceptionFrame,
+            ]
+        ],
+        outputdestination: str,
+        title: str = "Instance Group",
+        overwrite: bool = False,
+    ):
+        r"""
+        Static method to take a list of `InstanceGroup`\ s and turns each one into an indivual page
+        
+        Given a list of InstanceGroups and an output directory of *outputdestination* along with 
+        a *title* for the pages the generator loops through each group and calls the internal 
+        method to create HTML pages. *overwrite* indicates is previously existing HTML files should
+        be overwritten or not
+        """
+        for x in range(len(instances)):  # pylint:disable=consider-using-enumerate
+            CurriculumToHtml._generate(
+                instance=instances[x],
+                outputdestination=f"{outputdestination}{title}{x}.html",
+                title=title + f" {x}",
+                overwrite=overwrite,
+            )
+
+    @staticmethod
+    def _generate(
+        instance: InstanceGroup[
+            HighLevelSemanticsSituation,
+            LinearizedDependencyTree,
+            DevelopmentalPrimitivePerceptionFrame,
+        ],
+        title: str,
+        overwrite: bool,
+        outputdestination: str,
+    ):
+        """
+        Internal generation method for individual instance groups into HTML pages
+
+        Given an `InstanceGroup` with a `HighLevelSemanticsSituation`,
+        `LinearizedDependencyTree`, and `DevelopmentalPrimitivePerceptionFrame` this function
+        creates an html page at the given *outputdestination* and *title*. If the file already
+        exists and *overwrite* is set to False an error is raised in execution. Each page turns an
+        instance group with each "instance" as an indiviudal section on the page.
+
+        No returns
```

Comment:
No need to specify this - the type signature will put it into the docs automatically